app/src/test/java/com/github/se/signify/model/stats/MockStatsRepositoryTest.kt

```diff
@@ -0,0 +1,252 @@
+              mockStatsRepository::updateWeeklyQuestStats,
+              initialStats.weeklyQuest,
+              initialStats.weeklyQuest + 1),
+          arrayOf(
+              "getCompletedChallengeStats",
+              "updateCompletedChallengeStats",
+              mockStatsRepository::getCompletedChallengeStats,
+              mockStatsRepository::updateCompletedChallengeStats,
+              initialStats.completedChallenge,
+              initialStats.completedChallenge + 1),
+          arrayOf(
+              "getCreatedChallengeStats",
+              "updateCreatedChallengeStats",
+              mockStatsRepository::getCreatedChallengeStats,
+              mockStatsRepository::updateCreatedChallengeStats,
+              initialStats.createdChallenge,
+              initialStats.createdChallenge + 1),
+          arrayOf(
+              "getWonChallengeStats",
+              "updateWonChallengeStats",
+              mockStatsRepository::getWonChallengeStats,
+              mockStatsRepository::updateWonChallengeStats,
+              initialStats.wonChallenge,
+              initialStats.wonChallenge + 1))
+    }
+  }
+
+  @Before
+  fun resetStats() {
+    mockStatsRepository.reset()
+    mockStatsRepository.setStatsForUser(userId, initialStats)
+  }
+
+  @Test
+  fun `init should call onSuccess if shouldSucceed is true`() {
+    mockStatsRepository.init {}
+    assertTrue(mockStatsRepository.wasMethodCalled("init"))
+  }
+
+  @Test
+  fun `init should not call onSuccess if shouldSucceed is false`() {
+    mockStatsRepository.shouldSucceed = false
+    mockStatsRepository.init { fail("This should not be run") }
+    assertTrue(mockStatsRepository.wasMethodCalled("init"))
+  }
+
+  @Test
+  fun `wasMethodCalled should return true for called methods`() {
+    mockStatsRepository.getEasyExerciseStats(userId, {}, {})
+    assertTrue(mockStatsRepository.wasMethodCalled("getEasyExerciseStats"))
+  }
+
+  @Test
+  fun `wasMethodCalled should return false for methods not called`() {
+    assertFalse(mockStatsRepository.wasMethodCalled("updateMediumExerciseStats"))
+  }
+
+  @Test
+  fun `getMethodCalls should return list of called methods`() {
+    mockStatsRepository.getEasyExerciseStats(userId, {}, {})
+    mockStatsRepository.updateEasyExerciseStats(userId, {}, {})
+    val methodCalls = mockStatsRepository.getMethodCalls()
+    assertEquals(listOf("getEasyExerciseStats", "updateEasyExerciseStats"), methodCalls)
+  }
+
+  @Test
+  fun `reset should clear stats and method calls`() {
+    mockStatsRepository.getEasyExerciseStats(userId, {}, {})
+    mockStatsRepository.updateEasyExerciseStats(userId, {}, {})
+
+    mockStatsRepository.reset()
+
+    assertTrue(mockStatsRepository.getMethodCalls().isEmpty())
+    assertTrue(mockStatsRepository.wasMethodCalled("getEasyExerciseStats").not())
+    assertTrue(mockStatsRepository.wasMethodCalled("updateEasyExerciseStats").not())
+  }
+
+  @Test
+  fun `getStat should return the correct value`() {
```

Comment:
I believe testing invalid or null user IDs is unnecessary for the mock since it's designed to simulate controlled responses. Should be ok to assume valid input here imo